### ISSUE TITLE
Run valgrind memcheck in CI

### ISSUE THREAD

**Issue by nopeslide** [alrevuelta/cONNXr#54]
I found a few memory errors with valgrind in the base code.
To prevent such things in the future I propose running all tests with valgrind

**Comment by alrevuelta** [comment]
Yes, this is something that was definitely missing. Which specific memory errors you had?

**Comment by nopeslide** [comment]
> Yes, this is something that was definitely missing. Which specific memory errors you had?

you duplicate a string when integrating the input into the model.
this string had a fixed allocation size of 50 and it seems some model had a longer name :D
I implemented some std string utilities (`strnlen`,`strdup`, `strndup`) to replace this allocation in #53.
